src/System.Net.Http/src/System/Net/Http/MessageProcessingHandler.cs

```diff
@@ -12,6 +13,22 @@ namespace System.Net.Http
 {
     public abstract class MessageProcessingHandler : DelegatingHandler
     {
+        // Private class used to capture the SendAsync state in
+        // a closure, while simultaneously avoiding a tuple allocation.
+        private class SendState : TaskCompletionSource<HttpResponseMessage>
```

Comment:
@justinvp I followed @stephentoub's precedent [here](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.CurlResponseMessage.cs#L454) in `CurlHandler` for this class, which didn't use `sealed` and didn't have a `CompletionSource` in the class name. In hindsight though, it looks like `CancelableReadState` (which extends `ReadState`) is `sealed` since it doesn't have any subclasses, so I guess you're right in that aspect.